js/main.js

```diff
@@ -0,0 +1,206 @@
+        $modalGenres.textContent += modalGenreTypes[i].name + ' ';
+        $modalTextWrap.appendChild($modalGenres);
+      }
+
+      const $modalEpisodes = document.createElement('p');
+      $modalEpisodes.setAttribute('class', 'modal-episodes');
+      $modalEpisodes.textContent = top25Data[i].episodes + ' eps';
+      $modalTextWrap.appendChild($modalEpisodes);
+
+      const $modalDuration = document.createElement('p');
+      $modalDuration.setAttribute('class', 'modal-duration');
+      $modalDuration.textContent = 'Duration: ' + top25Data[i].duration;
+      $modalTextWrap.appendChild($modalDuration);
+
+      const $synopsisWrap = document.createElement('div');
+      $synopsisWrap.setAttribute('class', 'synopsis-wrap');
+      $modalContainer.appendChild($synopsisWrap);
+
+      const $synopsisHeading = document.createElement('h3');
+      $synopsisHeading.setAttribute('class', 'synopsis-heading');
+      $synopsisHeading.textContent = 'Synopsis';
+      $synopsisWrap.appendChild($synopsisHeading);
+
+      const $synopsis = document.createElement('p');
+      $synopsis.setAttribute('class', 'synopsis');
+      $synopsis.textContent = top25Data[i].synopsis;
+      $synopsisWrap.appendChild($synopsis);
+
+      const $trailerDiv = document.createElement('div');
+      $trailerDiv.setAttribute('class', 'trailer-wrap');
+      $modalContainer.appendChild($trailerDiv);
+
+      const $trailer = document.createElement('iframe');
+      $trailer.setAttribute('class', 'trailer');
+      $trailer.setAttribute('src', top25Data[i].trailer.embed_url);
+      $trailer.setAttribute('frameborder', '0');
+      $trailer.setAttribute('allow', 'accelerometer, clipboard-write; encrypted-media; gyroscope; picture-in-picture'
+      );
+      $trailer.setAttribute('allowfullscreen', 'allowfullscreen');
+      $trailerDiv.appendChild($trailer);
+
+      const $watchNowDiv = document.createElement('div');
+      $watchNowDiv.setAttribute('class', 'watch-now-wrap');
+      $modalContainer.appendChild($watchNowDiv);
+
+      const $watchNow = document.createElement('a');
+      $watchNow.textContent = 'Watch Now';
+      const websiteLink = top25Data[i].url;
+      $watchNow.setAttribute('href', websiteLink);
+      $watchNow.setAttribute('class', 'watch-now');
+      $watchNowDiv.appendChild($watchNow);
+
+      const $modalBtnWrap = document.createElement('div');
+      $modalBtnWrap.setAttribute('class', 'modal-btn-wrap');
+      $modalContainer.appendChild($modalBtnWrap);
+
+      const $modalAddBtn = document.createElement('button');
+      $modalAddBtn.textContent = '+ Add';
+      $modalAddBtn.setAttribute('class', 'add modal-btn');
+      $modalBtnWrap.appendChild($modalAddBtn);
+
+      const $modalCloseBtn = document.createElement('button');
+      $modalCloseBtn.textContent = 'Close';
+      $modalCloseBtn.setAttribute('class', 'close modal-btn');
+      $modalBtnWrap.appendChild($modalCloseBtn);
+      // end of modal elements
+    }
```

Comment:
This should all be moved into it's own function, rather than all in this `load` event listener.

Reply:
moved outside to its own function hiddenModalDOM(); 
called inside the load event listener